flambe/nlp/classification/task.py

```diff
@@ -0,0 +1,204 @@
+                 model: Optional[TextClassifier] = None,
+                 **kwargs) -> None:
+        """Initalize a TextClassification task.
+
+        Parameters
+        ----------
+        dataset : TabularDataset
+            The input dataset
+        encoder : Encoder
+            An encoder
+        pooling : Optional[Module], optional
+            A pooling method
+        dropout : float, optional
+            Dropout to apply between the encoder and output layer.
+            Default ``0``.
+        build_vocabularies : bool, optional
+            Whether the fields should expand their vocabulary Using
+            the training data. Default ``True``.
+        train_batch_size : int, optional
+            The batch size to use during. Default ``32``.
+        val_batch_size : int, optional
+            The batch size to use during evaluation. Default ``32``.
+        embedding_args : Dict[str, Any], optional
+            Keyword arguments to pass the ``Embeddings`` constructor.
+        optimizer : Optional[Optimizer], optional
+            The optimizer to use. Should be provided for training.
+        iter_scheduler : Optional[LRScheduler], optional
+            A learning rate scheduler to call on every training step.
+        eval_scheduler : Optional[LRScheduler], optional
+            A learning rate scheduler to call on every validation step.
+        text_field : Optional[TextField], optional
+            A custom text field to apply to the text inputs.
+        label_field : Optional[LabelField], optional
+            A custom label field to apply to the label inputs.
+        embedder : Optional[Embedder], optional
+            A custom embedder. Overrides ``encoder`` and ``pooling``.
+        model : Optional[TextClassifier], optional
+            A custom model. Overrides ``encoder`` and ``embedder``.
+
+        See the ``Training`` parent class for other keyword arguments.
+
+        """
+        super().__init__(**kwargs)
+
+        text_field = text_field or TextField()
+        label_field = label_field or LabelField()
+
+        # Build vocabularies
+        if build_vocabularies:
+            text, label = zip(*dataset)
+            text_field.setup(text)
+            label_field.setup(label)
+
+        self.embedding_args = embedding_args or dict()
+
+        self.train_batch_size = train_batch_size
+        self.val_batch_size = val_batch_size
+
+        self.dataset = dataset
+        self.text_field = text_field
+        self.label_field = label_field
+        self.dataset._set_transforms({'text': text_field, 'label': label_field}, do_setup=False)
+
+        if encoder is None and embedder is None and model is None:
+            raise ValueError("At least one of encoder, embedder or model must be provided.")
+        if model is None and embedder is None and encoder is not None and pooling is None:
+            raise ValueError("Must provide a pooling stratgey.")
```

Comment:
The factories would avoid a lot of these checks

Reply:
that's fair